crates/ruff/src/rules/flake8_slots/rules/no_slots_in_namedtuple_subclass.rs

```diff
@@ -0,0 +1,132 @@
+///
+/// ## Example
+/// ```python
+/// from collections import namedtuple
+///
+///
+/// class Foo(namedtuple("foo", ["name", "age"])):
+///     pass
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from collections import namedtuple
+///
+///
+/// class Foo(namedtuple("foo", ["name", "age"])):
+///     __slots__ = ()
+/// ```
+/// ## References
+/// - [Python Docs](https://docs.python.org/3.7/reference/datamodel.html#slots)
+/// - [StackOverflow](https://stackoverflow.com/questions/472000/usage-of-slots)
+#[violation]
+pub struct NoSlotsInNamedtupleSubclass;
+
+impl Violation for NoSlotsInNamedtupleSubclass {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Subclasses of `collections.namedtuple()` should define `__slots__`")
+    }
+}
+
+/// SLOT002
+pub(crate) fn no_slots_in_namedtuple_subclass<F>(
+    checker: &mut Checker,
+    class: &StmtClassDef,
+    locate: F,
+) where
+    F: FnOnce() -> TextRange + Copy,
+{
+    match class.bases.as_slice() {
+        [Expr::Name(ast::ExprName { id, .. })] => {
+            let scope = checker.semantic_model().scope();
+            if let Some(binding_id) = scope.get(id) {
+                let binding = &checker.semantic_model().bindings[binding_id];
+                if binding.kind.is_assignment() || binding.kind.is_named_expr_assignment() {
+                    if let Some(parent) = binding.source {
+                        let parent = checker.semantic_model().stmts[parent];
+                        match parent {
+                            Stmt::Assign(ast::StmtAssign { value, .. }) => {
+                                if is_indirect_namedtuple_subclass(checker, value.as_ref()) {
+                                    if !has_slots(&class.body) {
+                                        checker.diagnostics.push(Diagnostic::new(
+                                            NoSlotsInNamedtupleSubclass,
+                                            locate(),
+                                        ));
+                                    }
+                                }
+                            }
+                            Stmt::AnnAssign(ast::StmtAnnAssign { value, .. }) => {
+                                if let Some(val) = value.as_ref() {
+                                    if is_indirect_namedtuple_subclass(checker, val) {
+                                        if !has_slots(&class.body) {
+                                            checker.diagnostics.push(Diagnostic::new(
+                                                NoSlotsInNamedtupleSubclass,
+                                                locate(),
+                                            ));
+                                        }
+                                    }
+                                }
+                            }
+                            _ => (),
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Does the originating plugin do this? As far as I can tell, it doesn't.

Reply:
Gonna simplify for now to bias towards merging, but happy to revisit and we'll have it here in the Git history if we want to restore.